diagnostics_tools/include/diagnostic_tools/sampled_statistics.h

```diff
@@ -5,31 +5,59 @@
 #ifndef _DIAGNOSTIC_TOOLS__SAMPLED_STATISTICS_H_
 #define _DIAGNOSTIC_TOOLS__SAMPLED_STATISTICS_H_
 
-namespace qna {
-namespace diagnostic_tools {
-
+namespace qna
+{
+namespace diagnostic_tools
+{
 template <typename T>
-class SampledStatistics {
+class SampledStatistics
+{
  public:
-  void reset() {
+
+  SampledStatistics (size_t windows_size){
+    windows_size_ = windows_size;
+  }
+  SampledStatistics (){
+    windows_size_ = 100;
+  }
+
+  void reset()
+  {
     average_ = T{};
     minimum_ = T{};
     maximum_ = T{};
     sample_count_ = 0;
+    accumulate_ = T{};
   }
 
-  void update(const T& sample) {
-    average_ = (sample + average_ * sample_count_) / (sample_count_ + 1);
-    if (sample_count_ > 0) {
-      if (maximum_ < sample) {
+  void update(const T& sample)
+  {
+    if (sample_count_ < windows_size_)
+    {
+      accumulate_ += sample;
+      average_ = accumulate_ / sample_count_;
+      sample_count_++;
+    }
+    else
+    {
+      accumulate_ = accumulate_ + sample - last_value_;
```

Comment:
@patapufate it doesn't seem like `last_value_` is ever set? It doesn't seem like you can do this without a circular queue either.

Reply:
Implemented with queue.